okhttp/src/jvmMain/kotlin/okhttp3/internal/connection/RouteSelector.kt

```diff
@@ -160,13 +160,15 @@ class RouteSelector(
     } else {
       eventListener.dnsStart(call, socketHost)
 
-      // Try each address for best behavior in mixed IPv4/IPv6 environments.
       val addresses = address.dns.lookup(socketHost)
       if (addresses.isEmpty()) {
         throw UnknownHostException("${address.dns} returned no addresses for $socketHost")
       }
 
-      eventListener.dnsEnd(call, socketHost, addresses)
+      // Try each address for best behavior in mixed IPv4/IPv6 environments.
+      val prioritisedAddresses = InetAddressOrder.reorder(addresses)
+
+      eventListener.dnsEnd(call, socketHost, prioritisedAddresses)
```

Comment:
I think we should notify the EventListener with exactly what the network returned? Seems more useful in debugging than returning our processed result?